pkg/tests/tasks/migration/migration_test.go

```diff
@@ -0,0 +1,149 @@
+		t.Cleanup(func() {
+			oc.DeleteResource(t, "", "Istio", "example")
+			oc.DeleteResource(t, "", "IstioCNI", "default")
+		})
+		defaults := ossm.DefaultClusterWideSMCP()
+		ossm.BasicSetup(t)
+		templ := `apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: {{ .Name }}
+spec:
+  version: {{ .Version }}
+  tracing:
+    type: None
+  security:
+    manageNetworkPolicy: false
+  gateways:
+    enabled: false
+  policy:
+    type: Istiod
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  mode: ClusterWide`
+		oc.ApplyTemplate(t, defaults.Namespace, templ, defaults)
+
+		oc.WaitSMCPReady(t, defaults.Namespace, defaults.Name)
+		oc.WaitSMMRReady(t, defaults.Namespace)
+		oc.Label(t, "", "Namespace", ns.Bookinfo, "istio-injection=enabled")
+		t.LogStep("Update SMMR to include bookinfo namespace")
+		// Wait for SMMR to include bookinfo
+		oc.DefaultOC.WaitFor(t, defaults.Namespace, "ServiceMeshMemberRoll", "default", `jsonpath='{.status.configuredMembers[?(@=="bookinfo")]}'`)
+
+		t.LogStep("Create 3.0 controlplane and IstioCNI")
+		istio := `apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: example
+spec:
+  namespace: istio-system
+  version: v1.24.1`
+		oc.ApplyString(t, "", istio)
+		oc.DefaultOC.WaitFor(t, "", "Istio", "example", "condition=Ready")
+		oc.CreateNamespace(t, "istio-cni")
+		istioCNI := `apiVersion: sailoperator.io/v1alpha1
+kind: IstioCNI
+metadata:
+  name: default
+spec:
+  namespace: istio-cni
+  version: v1.24.1`
+		oc.ApplyString(t, "", istioCNI)
+
+		t.LogStep("Install bookinfo and bookinfo gateway")
+		app.InstallAndWaitReady(t, app.Bookinfo(ns.Bookinfo))
+
+		oc.ApplyFile(t, ns.Bookinfo, "bookinfo-gateway.yaml")
+		oc.DefaultOC.WaitDeploymentRolloutComplete(t, "bookinfo", "bookinfo-gateway")
+		ip := oc.DefaultOC.Invokef(t, "kubectl get service -n %s %s -o jsonpath='{.status.loadBalancer.ingress[0].ip}'", ns.Bookinfo, "bookinfo-gateway")
+		bookinfoGatewayURL := fmt.Sprintf("http://%s/productpage", ip)
+		curl.Request(t, bookinfoGatewayURL, nil, assert.RequestSucceeds("productpage request succeeded", "productpage request failed"))
+
+		t.LogStep("Migrate bookinfo to 3.0 controlplane")
+		oc.Label(t, "", "Namespace", ns.Bookinfo, "istio-injection- istio.io/rev=example")
+		// Wait for book info to be removed.
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			var members []string
+			output := oc.GetJson(t, defaults.Namespace, "ServiceMeshMemberRoll", "default", "{.status.configuredMembers}")
+			if err := json.Unmarshal([]byte(output), &members); err != nil {
+				t.Error(err)
+			}
+			contains := false
+			for _, member := range members {
+				if member == "bookinfo" {
+					contains = true
+					break
+				}
+			}
+			if contains {
+				t.Error("bookinfo found in SMMR. Expected it to be removed.")
+			}
+		})
```

Comment:
Ideally we should have constant load on the product page and check there are no failed requests during migration. That can be done in some more advanced test but let's add one more request here to verify it's still accessible when `istio-injection` is gone and it's not yet restarted?

Reply:
I added a function `continuallyRequest` which runs in the background of this test and will make a curl request every second to the bookinfo gateway url.